sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -2291,9 +2368,26 @@ pi_result piEventsWait(pi_uint32 NumEvents, const pi_event *EventList) {
     // NOTE: we are destroying associated command lists here to free
     // resources sooner in case RT is not calling piEventRelease soon enough.
     if (EventList[I]->ZeCommandList) {
-      // Event has been signaled: Destroy the command list associated with the
-      // call that generated the event.
-      ZE_CALL(zeCommandListDestroy(EventList[I]->ZeCommandList));
+      // Event has been signaled: If the fence for the associated command list
+      // is signalled, then reset the fence and command list and add them to the
+      // available list for ruse in PI calls.
+      EventList[I]->Queue->Context->Device->ZeCommandListFenceMapMutex.lock();
+      ZeResult = zeFenceQueryStatus(
```

Comment:
I wonder if this is at all possible for the "fence" not be signaled if the event is signaled already. Previously we were just destroying the L0 command-list here, so it must be completed, right? So, do we need to query the "fence" status here?